[Web] Refactor Back-End

moaikang opened the pull request:
### 🍞 Issue Number 

close: #61 
<br/>

### 🥞 작업 내역

> 구현 내용 및 작업 했던 내역

- [X] Type 이름 카멜케이스에서 파스칼 케이스로 변경
- [x] validate 로직 분리
- [x] 기타 등등 로직 분리
      <br/>

### 🍞 작업 유형

- [ ] 신규 기능 추가
- [ ] 버그 수정
- [x] 리펙토링
- [ ] 문서 업데이트
  <br/>

### 🥞 체크리스트

- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?
  <br/>

### 🥞 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 코드 보시고 이건 기존의 것이 더 낫겠다 싶은것이 있으면 말씀 주세요! 감사합니다!

<br/><br/>

Commit:
[Web] Refactor: :rainbow: refactor back-end type naming #61

백엔드 타입의 이름을 카멜 케이스에서 파스칼 케이스로 수정하였다.

Commit:
[Web] Refactor: :rainbow: change imported types' name #61

타입 이름을 파스칼 케이스로 바꾸었기 때문에, 이를 import해서 사용하는 곳의 타입 이름도 바꾸었다.

Commit:
[Web] Refactor: :rainbow: seperate roomCode validation logic from routes  #61

roomCode를 검증하는 로직을 utils로 분리하였다.

Commit:
[Web] Refactor: :rainbow: make router format equal in join #61

join router의 포맷을 똑같게 리팩토링하였다.

Commit:
[Web] Refactor: :rainbow: seperate randomCode logic from room router #61

랜덤한 룸 코드를 얻는 로직을 라우터로 부터 분리시켜 라우터를 가볍게 만들었다.

Commit:
[Web] Refactor: :rainbow: refactor status code to enum #61

Status Code를 enum 형태로 리팩토링하여 매직넘버를 없애고, 의미를 파악하기 쉽게 만들었다.

Commit:
[Web] Refactor: :rainbow: seperate socketLogin from socketController #61

소켓 컨트롤러의 로직을 소켓 서비스로 분리하였다.
그리고 자주 쓰이는 roomCode 만드는 함수와, roomCode로 부터 participants를 얻어내는 함수를 Util로 분리하였다.

Commit:
[Web] Refactor: :rainbow: seperate socketServerOption #61

socket Server Option을 다른 객체로 분리하였다.

Commit:
[Web] Refactor: :rainbow: Seperate logic of get now #61

현재 시간을 얻는 함수를 분리하였다.

Commit:
[Web] Refactor: :rainbow: change import function's file name #61

roomCode.ts를 room.ts로 이름을 바꾸었기 때문에, import roomcode에서 import room으로 선언부의 파일 이름이 바뀌었다.

Commit:
[Web] Fix : :fire: fix setRoom client method from hset to hmset #61

hset으로 사용하면 하나의 필드만 설정할 수 있기 때문에 계속 error가 발생했다.
이를 해결하기 위해 hset 메서드를 hmset 메서드로 수정하였다.

Commit:
[Web] Refactor: :rainbow: refactor rooms router's iteration from for-of to map #61

room 라우터에서 for of 로 비동기 처리를 하던 구문을 Promise.all과 map을 사용해 리팩토링했다.

Commit:
[Web] Refactor: :rainbow: Remove useless type in socketService #61

socketService에서 사용하지 않는 타입인 participantType을 제거하였다.

pje1740 commented on server/src/routes/join.ts:
```diff
@@ -1,33 +1,33 @@
-    }
+  const { roomCode } = req.body;
+
+  if (await isRoomCodeValidate(roomCode, 'private')) {
```
👍

pje1740 commented on server/src/routes/join.ts:
```diff
@@ -1,33 +1,33 @@
   }
-  return res.status(400).json();
+
+  return res.status(StatusCode.CLIENT_ERROR).json();
```
💯

pje1740 commented on server/src/socket/socket.ts:
```diff
@@ -1,25 +1,26 @@
-  });
+  };
+
+  const io = new socketIO.Server(server, socketServerOption);
```
옵션 이렇게 뺀 것도 좋은 거 같아요!

pje1740 commented on server/src/socket/socketController.ts:
```diff
@@ -1,74 +1,26 @@
-    participantsList: participantsList,
-    type: 'enter',
-  };
+  const renewedParticipants = await socketService.enterChatRoom(socket.id, roomCode, nickname, language);
```
앞에 객체로 묶어주긴 하지만, 이름이 감싸고 있는 함수랑 이름이 동일해서 혼동될 수 있을 거 같아요!
또 엄밀하게는 갱신된 리스트만 뽑아주는 함수인데 이걸 disconnect, enterChatRoom이라는 이름으로 두 번 작성된 느낌이라, 아예 갱신된 리스트를 뽑아주는 공통의 함수를 하나 만들고, type에 들어갈 값만 인자로 넘기면 어떨까요??

pje1740 commented on server/src/socket/socketService.ts:
```diff
@@ -0,0 +1,63 @@
+    type: 'enter',
+  };
+
+  return renewedParticipants;
```
위에 쓴 거랑 같은 맥락일 거 같긴한데 방에 소켓을 추가하고, 갱신된 리스트를 뽑는 걸 하나의 함수에서 처리하는 대신, 컨트롤러 쪽에서 끊어서 작성할 수 있으면 어떨까 싶네요!

pje1740 reviewed (commented):
소켓 컨트롤러 부분만 관련해서 의견 나눠보고 합의된 방향이 나오면 바로 merge해도 좋을 거 같습니다!! 주말에 혼자 고생하셨어요ㅠㅠㅠ

Commit:
[Web] Refactor: :rainbow: refactor with teammates #61

팀원과 함께 Refactor을 진행했다.
1. controller 로직 분리
2. util 함수 네이밍 통일
3. 외부 모듈과 자체제작 모듈 import 구분

kas136 reviewed (approved):
백엔드는 완성인가요 ㅠㅠㅠㅠㅠ

pje1740 reviewed (approved):
LGTM!!

moaikang commented:
너무 좋아요! 우리 이제 백엔드 끝!
